djautotask/sync_rest.py

```diff
@@ -689,3 +689,87 @@ def _assign_field_data(self, instance, object_data):
     def get_page(self, next_url=None, *args, **kwargs):
         kwargs['conditions'] = self.api_conditions
         return self.client.get_projects(next_url, *args, **kwargs)
+
+
+###################################################################
+# Dummy Synchronizers                                             #
+###################################################################
+
+
+class DummySynchronizer:
+    # Use FIELDS to list fields we submit to create or update a record, used
+    # as a kind of validation method and way to link the snake_case field
+    # names to their camelCase api names
+    FIELDS = {}
+
+    def __init__(self):
+        self.api_conditions = []
+        self.client = self.client_class()
+
+    def get(self, parent=None, conditions=None):
+
+        records = []
+        next_url = None
+
+        while True:
+            logger.info(
+                'Fetching {} records'.format(self.record_name)
+            )
+            conditions = conditions if conditions else self.api_conditions
+            api_return = self._get_page(
+                next_url, conditions, parent=parent)
+            records += api_return.get("items")
+            next_url = api_return.get("pageDetails").get("nextPageUrl")
+
+            if not next_url:
+                break
+
+        return records
+
+    def update(self, parent=None, **kwargs):
+        raise NotImplementedError
+
+    def create(self, parent=None, **kwargs):
+        raise NotImplementedError
+
+    def _get_page(self, next_url, conditions, parent=None):
+        raise NotImplementedError
+
+    def _format_record(self, **kwargs):
+        record = {}
+        for key, value in kwargs.items():
+            # Only consider fields of the record, discard anything else
+            if key in self.FIELDS.keys():
+                record[self.FIELDS[key]] = value
+
+        return record
+
+
+class TicketChecklistItemsSynchronizer(DummySynchronizer):
+    FIELDS = {
+        'id': 'id',
+        'ticket': 'ticketID',
+        'item_name': 'itemName',
+        'important': 'isImportant',
+        'completed': 'isCompleted',
+    }
+    client_class = api.TicketChecklistItemsAPIClient
+    record_name = "TicketChecklistItems"
+
+    # https://webservices2.autotask.net/atservicesrest/V1.0/Tickets/9980/ChecklistItems
+    def update(self, parent=None, **kwargs):
+        record = self._format_record(**kwargs)
+
+        return self.client.update(parent, **record)
+
+    def create(self, parent=None, **kwargs):
+        record = self._format_record(**kwargs)
+
+        return self.client.create(parent, **record)
+
+    def _get_page(self, next_url, conditions, parent=None):
+        if parent:
```

Comment:
It doesn't look like the parent conditions are applied when a parent value is passed in. I'm assuming parent is a TicketID in this case but maybe I'm just wrong about that?

![image](https://user-images.githubusercontent.com/44041923/124024600-43c85b00-d9a4-11eb-97fe-7fb3d5df7ab3.png)

Reply:
Hm, I'm not sure what is happening for you, it is working for me:

```
In [3]: sync.get(parent=9980)
[2021-06-30 14:10:49] INFO [djautotask.sync_rest:716] Fetching TicketChecklistItems records
[2021-06-30 14:10:49] DEBUG [djautotask.api_rest:352] Making GET request to https://webservices2.autotask.net/ATServicesRest/v1.0/ChecklistItems/query?search={"filter": [{"op": "eq", "field": "ticketID", "value": 9980}]}
```

Reply:
Oh I know why it wasn't working for me. I changed `TicketChecklistItemsAPIClient`'s API to `TicketChecklistItems` to test that that was the problem with the GET requests.